tests/graceful-recovery.md

```diff
@@ -0,0 +1,97 @@
+9. Inside the nginx container, navigate to `/etc/nginx/conf.d` and ensure that
+`http.conf` and `config-version.conf` look correct.
+10. In a different terminal, deploy the
+[https-termination example](https://github.com/nginxinc/nginx-gateway-fabric/tree/main/examples/https-termination).
+11. Inside the nginx container, check `http.conf` and `config-version.config` to see
+if the configuration and version were correctly updated.
+12. Send traffic through the example application and ensure it is working correctly.
+
+## Testing when nginx-gateway container restarts
+
+1. Ensure NGF and nginx container logs are set up and traffic flows through the example application correctly.
+2. Insert ephemeral container in NGF Pod and kill the nginx-gateway process.
+    1. `kubectl debug -it -n nginx-gateway <NGF_POD> --image=busybox:1.28 --target=nginx-gateway`
+    2. run `ps -A`
+    3. run `kill <nginx-gateway_PID>` (Command should start with `/usr/bin/gateway`)
+3. Check for errors in the NGF and nginx-container logs.
+4. When the nginx-gateway container is back up, ensure traffic flows through the example application correctly.
+5. Open up the NGF and nginx container logs and check for errors.
+6. Inside the nginx container, check that `http.conf` was not changed and `config-version.conf` had its version set to `2`.
+7. Send traffic through the example application and ensure it is working correctly.
+8. Check that NGF can still update statuses of resources.
+   1. Delete the HTTPRoute resources by running `kubectl delete -f cafe-routes.yaml` in `/examples/https-termination`
+   2. Inside the terminal which is inside the nginx container, check that `http.conf` and
+   `config-version.conf` were correctly updated.
+   3. Send traffic through the example application using the updated resources and ensure traffic does not flow.
+   4. Apply the HTTPRoute resources by running `kubectl apply -f cafe-routes.yaml` in `/examples/https-termination`
+   5. Inside the terminal which is inside the nginx container, check that `http.conf` and
+   `config-version.conf` were correctly updated.
+   6. Send traffic through the example application using the updated resources and ensure traffic flows correctly.
+
+## Testing when nginx container restarts
+
+1. Ensure NGF and nginx container logs are set up and traffic flows through the example application correctly.
+2. Insert ephemeral container in NGF Pod and kill the nginx-master process.
+   1. If there isn't already an ephemeral container inserted, run:
+   `kubectl debug -it -n nginx-gateway <NGF_POD> --image=busybox:1.28 --target=nginx-gateway`
+   2. run `ps -A`
+   3. run `kill <nginx-master_PID>` (Command should start with `nginx: master process`)
+3. When nginx container is back up, ensure traffic flows through the example application correctly.
+4. Open up the nginx-container logs and check for errors.
+5. Exec back into the nginx container and check that `http.conf` and `config-version.conf` were not changed.
```

Comment:
It will benefit if here we to repeat the steps similar to `8. Check that NGF can still update statuses of resources.` - to make sure NGF can reload NGINX in the new NGINX container.